### PR TITLE
ENH: Add alternate threshold effect

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
New threshold effect is a subclass of the threshold effect in the Slicer core that implements a new flood filling type of threshold.
Once the user has selected a threshold range + optional roi, they can ctrl+click on the image to add the region under the mouse to the image.

The voxels added to the segment are calculated using one of three methods:
* Masking
* GrowCut
* WaterShed

### CODE REVIEW

File: SegmentEditorThreshold2/SegmentEditorThreshold2Lib/SegmentEditorEffect.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    if os.path.exists(iconPath):
+      return qt.QIcon(iconPath)
+    return qt.QIcon()
+
+  def helpText(self):
+    return """<html>Fill segment based on master volume intensity range<br>. Options:<p>
+<ul style="feature: 0">
+<li><b>Use for masking:</b> set the selected intensity range as <dfn>Editable intensity range</dfn> and switch to Paint effect.</li>
+<li><b>Apply:</b> set the previewed segmentation in the selected segment. Previous contents of the segment is overwritten.</li>
+</ul><p></html>"""
```
Comment (lassoan) [inline SegmentEditorThreshold2/SegmentEditorThreshold2Lib/SegmentEditorEffect.py]: Add some info here about how to use it (control+click, ...)

File: CMakeLists.txt [hunk 2]
```diff
@@ -30,6 +30,7 @@ add_subdirectory(SegmentEditorSurfaceCut)
 add_subdirectory(SegmentEditorWatershed)
 add_subdirectory(SegmentEditorDrawTube)
 add_subdirectory(SegmentEditorSplitVolume)
+add_subdirectory(SegmentEditorThreshold2)
```
Comment (lassoan) [inline CMakeLists.txt]: "LocalThreshold" may be a better name.

**Comment by Sunderlandkyl** [comment]
- Renamed to Local Threshold.
- Reduced opacity of preview
- Updated help text
- New icon

**Comment by lassoan** [comment]
Does this work with Slicer-4.10?

**Comment by Sunderlandkyl** [comment]
Yes

**Comment by lassoan** [comment]
Perfect, thank you. Merged.

**Pull request merged**